doc/source/development/debugging_extensions.rst

```diff
@@ -0,0 +1,37 @@
+.. _debugging_c_extensions:
+
+{{ header }}
+
+**********************
+Debugging C extensions
+**********************
+
+Pandas uses select C extensions for high performance IO operations. In case you need to debug segfaults or general issues with those extensions, the following steps may be helpful. These steps are geared towards using lldb as a debugger, though the steps for gdb will be similar.
+
+First, be sure to compile the extensions with the appropriate flags to generate debug symbols and remove optimizations. This can be achieved as follows:
+
+.. code-block:: sh
+
+   python setup.py build_ext --inplace -j4 --with-debugging-symbols
+
+Next you can create a script that hits the extension module you are looking to debug and place it in the project root. Thereafter launch a Python process under lldb:
+
+.. code-block:: sh
+
+   lldb run python
+
+If desired, set breakpoints at various file locations using the below syntax:
+
+.. code-block:: sh
+
+   breakpoint set --file pandas/_libs/src/ujson/python/objToJSON.c --line 1547
+
+At this point you may get *WARNING:  Unable to resolve breakpoint to any actual locations.*. If you have not yet executed anything it is possible that this module has not been loaded into memory, which is why the location cannot be resolved. You can simply ignore for now as it will bind when we actually execute code.
+
+Finally go ahead and execute your script:
+
+.. code-block:: sh
+
+   run <the_script>.py
```

Comment:
what if i want to just run the test suite? (e.g. in the case of #35077 the tests that segfault only do so if run in the whole test suite, not something ive been able to replicate in the REPL)

Reply:
Does `run pytest` work?

Reply:
Or you can try `gdb --args python -m pytest ...` / `lldb -- python -m pytest` . the latter is working for me on my Mac

I don't totally understand how the whole launching / run machinery works just yet so there's probably another way to do it interactively

Reply:
Appended some info for this which should help